### ISSUE TITLE
Why does Scorex use custom serialization and de-serialization?

### ISSUE THREAD

**Issue by ceilican** [ScorexFoundation/Scorex#222]
While working on #221 I noticed that most of the uses of `Array` happen inside Scorex's custom (de)serializers. Moreover, most of the (de)serializers are very similar to each other. That's a lot of boilerplate and code duplication.

Why does Scorex have custom (de)serializers?  Why don't we use a standard (de)serialization or (de)serilization library? (cf. https://gist.github.com/ramn/5566596, http://lampwww.epfl.ch/~hmiller/pickling/, https://scalapb.github.io/generated-code.html, https://medium.com/@dkomanov/scala-serialization-419d175c888a)

**Comment by mslinn** [comment]
I agree on the value of using standard library for serialization. Pickling did not become popular, despite the strong initial push it got. Chill/kyro might be of interest https://github.com/twitter/chill
